### PR TITLE
Javadoc cleanup for new Application, Descriptor, Context and Table APIs.

### PR DESCRIPTION
**Pull request by prateekm**

### CODE REVIEW

**Comment by prateekm** [comment]
@vjagadish1989 @cameronlee314 @nickpan47 Please take a look. Would prefer to merge this in time for 1.0 (unless there are correctness issues) and fix forward with any minor changes.

**Comment by prateekm** [comment]
PS: I'd strongly prefer to disable checkstyle unused imports checks that get triggered due to javadoc references in samza-api. The fully qualified class names make the docs difficult to read. Will do that in a follow up PR after 1.0.

**Comment by cameronlee314** [comment]
@prateekm Could you please clarify what you will do in the follow up PR? I'm a bit unclear on what the proposed long-term state is.

**Comment by cameronlee314** [comment]
Oh, I think I see what you are proposing now. So for this PR, you have the fully qualified class names, but your follow up is to remove the fully qualified class names and then disable checkstyle. Is that right?

File: samza-api/src/main/java/org/apache/samza/application/SamzaApplication.java [hunk 1]
```diff
@@ -23,19 +23,29 @@
...
- * written in high-level DAG and low-level task APIs, respectively.
+ * This is the base {@link SamzaApplication}. Implement a {@link StreamApplication} to describe the
+ * processing logic using Samza's High Level API in terms of {@link org.apache.samza.operators.MessageStream}
+ * operators, or a {@link TaskApplication} to describe it using Samza's Low Level API in terms of per-message
+ * processing logic.
+ * <p>
+ * A {@link SamzaApplication} implementation must have a no-argument constructor, which will be used by the framework
+ * to create new instances and call {@link #describe(ApplicationDescriptor)}.
+ * <p>
+ * Per container context may be managed using {@link org.apache.samza.context.ApplicationContainerContext} and
```
[the remark this answers is not in the document]
Comment (prateekm) [reply samza-api/src/main/java/org/apache/samza/application/SamzaApplication.java]: Updated the other sentence to refer to ApplicationTaskContext instead.

**Pull request closed without merging**